cl/opinion_page/forms.py

```diff
@@ -78,3 +96,201 @@ class DocketEntryFilterForm(forms.Form):
+            )
+            del self.fields["second_judge"]
+            del self.fields["third_judge"]
+
+        else:
+            self.fields["cite_reporter"].widget = forms.Select(
+                choices=[("TN WC App.", "TN WC App.")],
+                attrs={"class": "form-control"},
+            )
+
+    court_str = forms.CharField(required=True, widget=forms.HiddenInput(),)
+
+    case_title = forms.CharField(
+        label="Case Title",
+        required=True,
+        widget=forms.TextInput(
+            attrs={
+                "class": "form-control",
+                "placeholder": "Case Title",
+                "autocomplete": "off",
+            }
+        ),
+    )
+
+    docket_number = forms.CharField(
+        label="Docket Number",
+        required=True,
+        widget=forms.TextInput(
+            attrs={
+                "class": "form-control",
+                "placeholder": "Docket Number",
+                "autocomplete": "off",
+            }
+        ),
+    )
+
+    publication_date = forms.DateField(
+        label="Publication Date",
+        required=True,
+        widget=forms.TextInput(
+            attrs={
+                "class": "form-control datepicker",
+                "placeholder": "Publication Date",
+                "autocomplete": "off",
+            }
+        ),
+    )
+
+    lead_author = forms.ModelChoiceField(
+        queryset=Person.objects.none(),
+        required=True,
+        widget=forms.Select(
+            attrs={
+                "class": "form-control",
+                "placeholder": "Choose lead Author",
+            }
+        ),
+    )
+
+    second_judge = forms.ModelChoiceField(
+        queryset=Person.objects.none(),
+        required=False,
+        label="Second panelist",
+        widget=forms.Select(attrs={"class": "form-control",}),
+    )
+
+    third_judge = forms.ModelChoiceField(
+        queryset=Person.objects.none(),
+        required=False,
+        label="Third panelist",
+        widget=forms.Select(attrs={"class": "form-control",}),
+    )
+
+    cite_volume = forms.IntegerField(
+        label="Cite year",
+        required=True,
+        widget=forms.Select(
+            choices=[(x, x) for x in xrange(datetime.now().year, 2013, -1)],
+            attrs={"class": "form-control"},
+        ),
+    )
+
+    cite_reporter = forms.CharField(label="Cite reporter", required=True,)
+
+    cite_page = forms.IntegerField(
+        label="Cite page",
+        required=True,
+        widget=forms.TextInput(
+            attrs={"class": "form-control", "autocomplete": "off",}
+        ),
+    )
+
+    pdf_upload = forms.FileField(
+        label="Opinion PDF",
+        required=True,
+        validators=[FileExtensionValidator(["pdf"])],
+    )
+
+    def validate_neutral_citation(self):
+        cv = self.cleaned_data.get("cite_volume")
+        cr = self.cleaned_data.get("cite_reporter")
+        cp = self.cleaned_data.get("cite_page")
```

Comment:
My knowledge of Django Forms is (growing) not broad.  I didn't know why to use .get or not.  I'll update the names for the variables.

Reply:
Well, `.get()` vs access by keyname isn't form related so much as style related. I'd never use a .get() when key access is reliable (which I think using cleaned_data guarantees. 

In general, if you're looking at code and see a `.get()` instead of a key-based access you have to wonder why the key could be missing. If it can't, that's just extra confusion, so I avoid `.get()`'s unless needed.